data_quality/tasks/aggregate.py

```diff
@@ -111,50 +141,129 @@ def get_source(self, data_src):
 
         return matches[0]
 
-    def get_pipeline_score(self, pipeline):
+    def get_pipeline_report_url(self, pipeline):
+        """Return a URL to a report on this data."""
+
+        return self.config['goodtables']['goodtables_web']
+
+    def get_pipeline_score(self, pipeline, source):
         """Return a score for this pipeline run."""
 
         score = self.max_score
-        results = pipeline.report.generate()['results']
-
-        if not results:
-            pass
+        report = pipeline.report.generate()
+        error_stats = self.get_error_stats(report)
+        total_no_rows = report['meta']['row_count']
+        base_errors = {err: stats for err, stats in error_stats.items()
+                       if stats['processor'] == 'base'}
+        if base_errors:
+            score = 0
         else:
-            result_count = len(results)
-
-            if result_count > score:
-                score = 0
+            if self.scoring_algorithm == 1:
+                score = self.score_by_affected_rows(error_stats, total_no_rows+1)
+            elif self.scoring_algorithm == 2:
+                score = self.score_by_error_occurences(error_stats)
             else:
-                score = score - result_count
-
+                score = self.score_by_affected_rows(error_stats, total_no_rows+1,
+                                                    weighted=True)
+            if self.assess_timeliness:
+                publication_delay = self.get_publication_delay(source)
+                score -= publication_delay
```

Comment:
I settled on subtracting 1 for each month of delay because treating it like a normal error in the scoring algorithms doesn't make much sense: in `score_by_error_occurrences`, the impact would decrease as more months of delay accumulate, which is not the case; in `score_by_affected_rows` the effect would be the same as this.